Sources/TSCLibc/libc.swift

```diff
@@ -75,24 +75,35 @@ public func mkstemps(
     _ template: UnsafeMutablePointer<CChar>?,
     _ suffixlen: Int32
 ) -> Int32 {
-  guard let template = template else { return -EINVAL }
-  return String(cString: template).withCString(encodedAs: UTF16.self) {
-    let capacity: Int = wcslen($0) + 1
-    return $0.withMemoryRebound(to: wchar_t.self, capacity: capacity) {
-      guard _wmktemp_s(UnsafeMutablePointer(mutating: $0), capacity) == 0 else {
-        return -EINVAL
-      }
-
-      var fd: Int32 = -1
-      _wsopen_s(&fd, $0, _O_RDWR | _O_CREAT | _O_BINARY | _O_NOINHERIT,
-                _SH_DENYNO, _S_IREAD | _S_IWRITE)
-
-      String(decodingCString: $0, as: UTF16.self).utf8CString.withUnsafeBytes {
-        template.assign(from: $0.bindMemory(to: CChar.self).baseAddress!,
-                        count: $0.count)
-      }
+  // Although the signature of the function is `char *(*)(char *)`, the C
+  // library treats it as `char *(*)(char * _Nonull)`.  Most implementations
+  // will simply use and trigger a segmentation fault on x86 (and similar faults
+  // on other architectures) when the memory is accessed.  This roughly emulates
+  // that by terminating in the case even though it is possible for us to return
+  // an error.
+  guard let template = template else { fatalError() }
+
+  let length: Int = strlen(template)
+
+  // Validate the precondition: the template must terminate with 6 `X` which
+  // will be filled in to generate a unique directory.
+  guard length >= 6, memcmp(template + length - Int(suffixlen) - 6, "XXXXXX", 6) == 0 else {
```

Comment:
Looks like this would crash if `suffixlen` is too large.  Or perhaps there's a guard that I'm just not seeing?

Reply:
If `suffixlen` is enough to wrap beyond the limits of `Int`, then the string buffer would have to be beyond 4GiB (though in this case beyond the addressing limits actually).  These functions are trying to emulate the C behavior.  If `suffixlen` is greater than `length` we would actually be accessing invalid memory.

Reply:
Right, I was talking about the case in which `suffixlen` is greater than the string length.  Since this is emulating C behavior, which all its caveats, it seems reasonable to have the same unsafety.

Reply:
Yeah, even tries to emulate the failure on a nullptr input :).  Hopefully we will soon be able to replace some of these unsafe functions with better alternatives.

Reply:
Right, I saw that.  Crash for crash compatible.  :)  Thanks for your work on this!  Please go ahead and merge after the tests complete.